common/proto/evm.token.proto

```diff
@@ -0,0 +1,45 @@
+syntax = "proto3";
+
+package messari.evm.token.v1;
+
+message Tokens {
+  repeated Token items = 1;
+}
+
+message Token {
+  string address = 1;
+  string name = 2;
+  string symbol = 3;
+  uint64 decimals = 4;
+  optional string total_supply = 5;
```

Comment:
I would separate `total_supply` from `Token` as all other fields are immutable. One way to do it is to represent it using `TokenBalance` instead.

Reply:
exactly

Reply:
Since total_supply can change do you want a block number associated with it? is that the thing?

The way I restructured the proto, it doesn't feel right to put it in TokenBalance

Reply:
If the block associated with the total supply is the issue, wouldn't we be able to get the block number on the block that  outputted that message in the output?

Reply:
I'd imagine `Token` (without total supply) is a very commonly needed/used data structure across many substreams, whereas total supply is used much less. If you include total supply in `Token`, every change to total supply will lead to a new message sent over to the downstream modules, which will lead to lots of unnecessary data and compute incurred to the downstream modules.

Reply:
For now I am removing it from the eth-balance proto, I think if we want to do that we should have a separate substream to calculate the total supply of eth. We could consider adding it into the erc20 substream if that is important.

Reply:
yeah, having a separate one sounds good to me. I actually had an issue open for this earlier: https://github.com/messari/substreams/issues/94